lib/ngx/log.lua

```diff
@@ -0,0 +1,83 @@
+
+local ffi = require 'ffi'
+local base = require "resty.core.base"
+local ffi_string = ffi.string
+local get_string_buf = base.get_string_buf
+local get_size_ptr = base.get_size_ptr
+local C = ffi.C
+local new_tab = base.new_tab
+local ffi_new = ffi.new
+local charpp = ffi_new("char *[1]")
+local intp = ffi.new("int[1]")
+
+
+local _M = { version = base.version }
+
+
+ffi.cdef[[
+    int ngx_http_lua_ffi_set_errlog_filter(int level, unsigned char *err,
+        size_t *errlen);
+    int ngx_http_lua_ffi_get_errlog_data(char **log, size_t *loglen,
+        int *loglevel, unsigned char *err, size_t *errlen);
```

Comment:
Let's remove the leading indentations to save some space in this Lua string literal.